pkg/networking/session_test.go

```diff
@@ -0,0 +1,483 @@
+package networking_test
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"io"
+	"log/slog"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/neilotoole/slogt"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+
+	"github.com/wavesplatform/gowaves/pkg/networking"
+	netmocks "github.com/wavesplatform/gowaves/pkg/networking/mocks"
+)
+
+func TestSuccessfulSession(t *testing.T) {
+	defer goleak.VerifyNone(t)
+
+	p := netmocks.NewMockProtocol(t)
+	p.On("EmptyHandshake").Return(&textHandshake{}, nil)
+	p.On("IsAcceptableHandshake", &textHandshake{v: "hello"}).Once().Return(true)
+	p.On("IsAcceptableHandshake", &textHandshake{v: "hello"}).Once().Return(true)
+	p.On("EmptyHeader").Return(&textHeader{}, nil)
+	p.On("IsAcceptableMessage", &textHeader{l: 2}).Once().Return(true)
+	p.On("IsAcceptableMessage", &textHeader{l: 13}).Once().Return(true)
+
+	clientHandler := netmocks.NewMockHandler(t)
+	serverHandler := netmocks.NewMockHandler(t)
```

Comment:
Why do you not assert expectations for mocks in all tests?
- `defer mockProtocol.AssertExpectations(t)`
- `defer clientHandler.AssertExpectations(t)`
- `defer serverHandler.AssertExpectations(t)`

Reply:
Found assertion inside mocks constructors in generated code
```go
func NewMockProtocol(t interface {
	mock.TestingT
	Cleanup(func())
}) *MockProtocol {
	mock := &MockProtocol{}
	mock.Mock.Test(t)

	t.Cleanup(func() { mock.AssertExpectations(t) })

	return mock
}
```